pkg/controller/plan/kubevirt.go

```diff
@@ -786,6 +806,23 @@ func (r *KubeVirt) emptyVm(vm *plan.VMStatus) (virtualMachine *cnv.VirtualMachin
 	return
 }
 
+// Create empty VM definition with origin name lable.
+func (r *KubeVirt) emptyVmwithOriginalNameLable(vm *plan.VMStatus, originalName string) (virtualMachine *cnv.VirtualMachine) {
```

Comment:
how about having it in a single place (e.g `emptyVm` above), getting also `originalName`. you can create the vm with `ObjectMeta` that doesn't contain the `Labels` and add it at the end based on the `if`. Another option would be to have 2 functions creating the `ObjectMeta`.

Reply:
good idea, changed